src/cls/rbd/cls_rbd.cc

```diff
@@ -2669,9 +2669,10 @@ int object_map_snap_add(cls_method_context_t hctx, bufferlist *in,
   }
 
   bool updated = false;
-  for (uint64_t i = 0; i < object_map.size(); ++i) {
-    if (object_map[i] == OBJECT_EXISTS) {
-      object_map[i] = OBJECT_EXISTS_CLEAN;
+  auto it = object_map.begin();
+  for (; it != object_map.end(); ++it) {
```

Comment:
Nit: store end iterator in a variable (a la ```auto end_it = object_map.end()```) to avoid recomputing its index each time.